rekall-core/rekall/plugins/core.py

```diff
@@ -958,13 +967,21 @@ def render(self, renderer):
             address_space=self.plugin_args.address_space,
             session=self.session)
 
+        count = 0
         for hit, _ in scanner.scan(offset=self.plugin_args.offset,
                                    maxlen=self.plugin_args.limit):
             hexdumper = self.session.plugins.dump(
-                offset=hit - 16, length=self.plugin_args.context + 16,
+                offset=hit - self.plugin_args.before, length=16 + self.plugin_args.after,
                 address_space=self.plugin_args.address_space)
 
             hexdumper.render(renderer)
+            count += 1
+
+            if count == self.plugin_args.max:
+                break
+
+        if self.plugin_args.count:
+            renderer.write("--\nFound {:,} hits.\n".format(count))
```

Comment:
We try to deprecate renderer.write() since it does not produce machine readable data. I dont think the count arg is useful since you can always just print it using an efilter query:

```
select count(*) from grep(....)
```

Alternatively you can add an extra hidden column with the hit count if you think this makes it easier? Then users can just include the count column in the output if they wanted it.

Reply:
Hi Michael,

Thanks for the feedback.

I agree that the non-standard (non machine readable) output makes it unfriendly.

I do think the hidden column of count is better, otherwise I have to run the query twice: once to see the results, once to count them.

I will look to implement this instead. Can you provide any pointers as to how I make a "hidden" column? Is this a particular thing within Rekall?

Thanks!

Reply:
I just noticed that grep is still an old style plugin (with a render method instead of collect). Its probably worth just making it look like the dump plugin just above it:

1. Add a table_header just like dump, with the extra column dict(name="count", hidden=True).
2. Change render() method to collect() method. In Grep you would still delegate to the dump plugin as before but instead of calling dumper.render() you would do something like:

```
class Grep....:
   # Same columns as dump but an extra column.
   table_header = [
     dict(name="count", hidden=True)
   ] + Dump.table_header

   def collect(self):
   ....
    dumper = ....
    for row in dumper.collect():
       row["count"] = count
       yield row
```

Reply:
Thanks Michael.

I think I'm almost there.

Question: how does one "see" the hidden rows?

Reply:
The hidden row is still selectable in efilter queries, or using the --verbosity 10 flag it will also show them.